org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/participants/codeactions/cvc_complex_type_4CodeAction.java

```diff
@@ -71,4 +76,19 @@ public void doCodeAction(Diagnostic diagnostic, Range range, DOMDocument documen
 		}
 	}
 
+	private boolean codeAlreadyActionExists(List<CodeAction> codeActions, Diagnostic diagnostic ) {
+		if(diagnostic == null) {
+			return false;

```

Comment:
null-check should go in doCodeAction. Because if it's really null, then you end up in L45, where it'll blow up